trio/_core/_io_kqueue.py

```diff
@@ -96,49 +105,54 @@ def process_events(self, events):
     # be more ergonomic...
 
     @_public
-    def current_kqueue(self):
+    def current_kqueue(self) -> select.kqueue:
         return self._kqueue
 
     @contextmanager
     @_public
-    def monitor_kevent(self, ident, filter):
+    def monitor_kevent(
+        self, ident: int, filter: int
+    ) -> Iterator[_core.UnboundedQueue[select.kevent]]:
         key = (ident, filter)
         if key in self._registered:
             raise _core.BusyResourceError(
                 "attempt to register multiple listeners for same ident/filter pair"
             )
-        q = _core.UnboundedQueue()
+        q = _core.UnboundedQueue[select.kevent]()
```

Comment:
I believe `monitor_kevent` is somewhat experimental/not fully tested and implemented, which is why it's failing coverage here.